File: Backend/src/main/java/lan/dk/podcastserver/manager/worker/updater/UploadUpdater.java
package lan.dk.podcastserver.manager.worker.updater;

import javaslang.collection.HashSet;
import javaslang.collection.Set;
import lan.dk.podcastserver.entity.Item;
import lan.dk.podcastserver.entity.Podcast;
import lan.dk.podcastserver.service.SignatureService;
import lan.dk.podcastserver.service.properties.PodcastServerParameters;
import org.springframework.stereotype.Component;

import javax.validation.Validator;
import java.util.function.Predicate;

/**
 * Created by kevin on 15/05/15 for HackerRank problem
 */
@Component("UploadUpdater")
public class UploadUpdater extends AbstractUpdater {

    UploadUpdater(PodcastServerParameters podcastServerParameters, SignatureService signatureService, Validator validator) {
        super(podcastServerParameters, signatureService, validator);
    }

    @Override
    public Set<Item> getItems(Podcast podcast) {
        return HashSet.ofAll(podcast.getItems());
    }

    @Override
    public String signatureOf(Podcast podcast) {
        return "";
    }

    @Override
    public Predicate<Item> notIn(Podcast podcast) {
        return item -> Boolean.FALSE;
    }

    @Override
    public Type type() {
        return new Type("upload", "Upload");
    }

    @Override
    public Integer compatibility(String url) {
        return Integer.MAX_VALUE;
    }
}

File: backend/src/main/kotlin/com/github/davinkevin/podcastserver/update/updaters/upload/UploadUpdater.kt
package com.github.davinkevin.podcastserver.update.updaters.upload

import com.github.davinkevin.podcastserver.update.updaters.ItemFromUpdate
import com.github.davinkevin.podcastserver.update.updaters.PodcastToUpdate
import com.github.davinkevin.podcastserver.update.updaters.Type
import com.github.davinkevin.podcastserver.update.updaters.Updater
import io.micrometer.core.instrument.MeterRegistry
import java.net.URI

class UploadUpdater(
    override val registry: MeterRegistry,
) : Updater {
    override fun findItems(podcast: PodcastToUpdate): List<ItemFromUpdate> = emptyList()
    override fun signatureOf(url: URI): String = ""
    override fun type(): Type = TYPE
    override fun compatibility(url: String): Int = Integer.MAX_VALUE
}

private val TYPE = Type("upload", "Upload")
